### DISCUSSION TITLE
problam with compiler/Linker while  using heap

### DISCUSSION THREAD

**Discussion by adlv888** [amichai-bd/riscv-multi-core-lotr#65] [Q&A]
while defining an array , if array smaller then 4 then all goes well :
in c:
![image](https://user-images.githubusercontent.com/81635109/121008494-bbd3a480-c79b-11eb-8dff-f22eca1da4e0.png)
after compile: 
![image](https://user-images.githubusercontent.com/81635109/121008568-ce4dde00-c79b-11eb-879d-073860594791.png)

but with larger arrays:
c:
![image](https://user-images.githubusercontent.com/81635109/121008690-efaeca00-c79b-11eb-8cde-420bbc41cc5c.png)
after compile:
![image](https://user-images.githubusercontent.com/81635109/121008741-fdfce600-c79b-11eb-9c7d-7f44e05b65e1.png)
no assigning the values

**Comment by amichai-bd** [comment]
@DL8

**Comment by DL8** [comment]
Heap is not the right term here: from what I saw in my experiments, a big array is pre-loaded into a different section in the ELF binary called ".rodata" (for the reference, ".text" is the section with the code). In regular app execution, this section is loaded into read-only memory. To fix this, the linker script should be fixed such that it will load the ".rodata" section to D_MEM (or find a way to force the compiler not to do this trick)
